libsnark/zk_proof_systems/plonk/prover.tcc

```diff
@@ -0,0 +1,1211 @@
+    printf("W_zeta_part[1]\n");
+    print_vector(W_zeta_part[1]);
+    printf("W_zeta_part[2]\n");
+    print_vector(W_zeta_part[2]);
+    printf("W_zeta_part[3]\n");
+    print_vector(W_zeta_part[3]);
+    printf("W_zeta_part[4]\n");
+    print_vector(W_zeta_part[4]);
+    printf("W_zeta_part[5]\n");
+    print_vector(W_zeta_part[5]);
+    printf("W_zeta_part[6]\n");
+    print_vector(W_zeta_part[6]);
+    printf("W_zeta\n");
+    print_vector(W_zeta);
+    printf("W_zeta_omega\n");
+    print_vector(W_zeta_omega);
+#endif // #ifdef DEBUG
+
+    assert(W_zeta == example.W_zeta);
+    assert(W_zeta_omega == example.W_zeta_omega);
+
+    // Evaluate polynomials W_zeta and W_zeta_omega at the seceret
+    // input
+    round_five_out.W_zeta_at_secret =
+        plonk_evaluate_poly_at_secret_G1<ppT>(srs.secret_powers_g1, W_zeta);
+    round_five_out.W_zeta_omega_at_secret =
+        plonk_evaluate_poly_at_secret_G1<ppT>(
+            srs.secret_powers_g1, W_zeta_omega);
+
+    // Hashes of transcript (Fiat-Shamir heuristic) -- fixed to match
+    // the test vectors
+    round_five_out.u = example.u;
+
+    return round_five_out;
+}
+
+// Prover compute SNARK proof
+//
+// Pi ([a]_1, [b]_1, [c]_1, [z]_1,
+//     [t_lo]_1, [t_mi]_1, [t_hi]_1,
+//     \bar{a}, \bar{b}, \bar{c},
+//     \bar{S_sigma1}, \bar{S_sigma2}, \bar{z_w},
+//     [W_zeta]_1, [W_{zeta omega}]_1
+//     r_zeta (*))
+//
+// (*) Note: in the reference Python implementation, r_zeta (the
+// evaluation of the linearlization polynomial r(X) at zeta from
+// Prover round 5) is added to the pi-SNARK proof. In the paper
+// this is omitted, which seems to make the proof shorter by 1
+// element at the epxense of a slightly heavier computation on the
+// verifier's side. Here we follow the reference implementation to
+// make sure we match the test values. TODO: once all test vectors
+// are verified, we may remove r_zeta from the proof to be fully
+// compliant with the paper.
+//
+// Mapping code-to-paper quantities
+//
+// - W_polys_blinded_at_secret_g1[a, b, c]: [a]_1, [b]_1, [c]_1 (from Round 1)
+// - z_poly_at_secret_g1: [z]_1 (from Round 2)
+// - t_poly_at_secret_g1[lo, mi, hi]: [t_lo]_1, [t_mi]_1, [t_hi]_1 (from Round
+// 3)
+// - a_zeta, b_zeta, c_zeta, S_0_zeta, S_1_zeta, z_poly_xomega_zeta: \bar{a},
+// \bar{b}, \bar{c}, \bar{S_sigma1}, \bar{S_sigma2}, \bar{z_w} (from Round 4)
+// - W_zeta_at_secret, W_zeta_omega_at_secret: [W_zeta]_1, [W_{zeta omega}]_1
+// (from Round 5)
+//
+// INPUT
+// - common_input: common preprocessed input
+// - srs: structured reference string
+//
+// OUTPUT
+// - proof: SNARK proof Pi (see above)
+//
+template<typename ppT>
+plonk_proof<ppT> plonk_prover<ppT>::compute_proof(
+    const srs<ppT> srs, const common_preprocessed_input<ppT> common_input)
+{
+    using Field = libff::Fr<ppT>;
+
+    // initialize hard-coded values from example circuit
+    plonk_example<ppT> example;
+    std::vector<Field> witness = example.witness;
+#ifdef DEBUG
```

Comment:
no need for ifdef guards here.

Reply:
This will be addressed when I remove all `#ifdef DEBUG`-s from all implementation files as discussed